packages/search-ui/src/__tests__/DebounceManager.test.js

```diff
@@ -0,0 +1,111 @@
+import DebounceManager from "../DebounceManager";
+
+describe("#runWithDebounce", () => {
+  it("will run the provided function after the provided number of milliseconds", () => {
+    const manager = new DebounceManager();
+    let debounced = jest.fn();
+
+    manager.runWithDebounce(1000, "debounced", () => {
+      debounced();
+    });
+
+    jest.advanceTimersByTime(500);
+    expect(debounced.mock.calls.length).toBe(0);
+    jest.advanceTimersByTime(1001);
+    expect(debounced.mock.calls.length).toBe(1);
+  });
+
+  it("will debounce the function", () => {
+    const manager = new DebounceManager();
+    let debounced = jest.fn();
+
+    manager.runWithDebounce(1000, "debounced", () => {
+      debounced();
+    });
+
+    jest.advanceTimersByTime(100);
+
+    manager.runWithDebounce(1000, "debounced", () => {
+      debounced();
+    });
+
+    expect(debounced).not.toHaveBeenCalled();
+    jest.advanceTimersByTime(1001);
+    expect(debounced.mock.calls.length).toBe(1);
+  });
+
+  it("will will debounce functions with different wait times separately", () => {
+    const manager = new DebounceManager();
+    let debounced = jest.fn();
+
+    manager.runWithDebounce(1000, "debounced", () => {
+      debounced();
+    });
+
+    manager.runWithDebounce(1000, "debounced", () => {
+      debounced();
+    });
+
+    manager.runWithDebounce(1000, "debounced", () => {
+      debounced();
+    });
+
+    manager.runWithDebounce(999, "debounced", () => {
+      debounced();
+    });
+
+    expect(debounced).not.toHaveBeenCalled();
+    jest.advanceTimersByTime(1001);
+    expect(debounced.mock.calls.length).toBe(2);
+  });
+
+  it("will will debounce functions with different names separately", () => {
+    const manager = new DebounceManager();
+    let debounced = jest.fn();
+
+    manager.runWithDebounce(1000, "debounced", () => {
+      debounced();
+    });
+
+    manager.runWithDebounce(1000, "debounced", () => {
+      debounced();
+    });
+
+    manager.runWithDebounce(1000, "different", () => {
+      debounced();
+    });
+
+    manager.runWithDebounce(999, "debounced", () => {
+      debounced();
+    });
+
+    expect(debounced).not.toHaveBeenCalled();
+    jest.advanceTimersByTime(1001);
+    expect(debounced.mock.calls.length).toBe(3);
+  });
+});
+
+describe("#cancelByName", () => {
+  const manager = new DebounceManager();
+  let debounced = jest.fn();
+
+  manager.runWithDebounce(1000, "debounced", () => {
+    debounced();
+  });
+
+  manager.runWithDebounce(500, "debounced", () => {
+    debounced();
+  });
+
+  manager.runWithDebounce(1000, "different", () => {
+    debounced();
+  });
+
+  manager.runWithDebounce(999, "much different", () => {
+    debounced();
+  });
+
+  manager.cancelByName("debounced");
+  jest.advanceTimersByTime(1001);
+  expect(debounced.mock.calls.length).toBe(2);
+});
```

Comment:
Do we need to write a test for `DebounceManager.debounce` or nah since it's really just calling a 3rd party dep?

Reply:
I didn't just because it was a simple pass-through. But I could for completeness.

Reply:
Depends on how much you care about line/function % coverage. I'm an overachieving completionist when it comes to that stuff but like you said, it's pretty straightforward in this case, so I'm fine with skipping for speed 👍